### PR TITLE
Dynamic favicon

### PR DESCRIPTION
**Pull request by DivineOmega**
Relates to issue #686.

The following logic dictates which icon is shown. Feedback on this is very welcome. :)

Image | Conditions
------------ | -------------
favicon-high-alert High alert favicon (red exclamation mark) | A component is experiencing a major outage (status 4) or an incident has been identified, but a fix has not yet been released (status 2).
favicon-medium-alert Medium alert favicon (yellow exclamation mark) | A component is experiencing performance issues or a partial outage (status 2 or 3) or an incident is being investigated or an incident fix is being watched (status 1 or 3).
favicon Regular favicon (Cachet logo) | None of the above conditions are met.

First code contribution to Cachet, so I'd love to hear your thoughts.

### CODE REVIEW

**Comment by TheoBearman** [comment]
@DivineOmega Personally I would have thought different coloured Cachet logos would have been the best way to display an error, not just a simple exclamation mark. Thoughts @jbrooksuk?

**Comment by jbrooksuk** [comment]
Let me think on this. Since Cachet only shows red/green it may be a little complicated.

In regards to what @TheoBearman said, I think it'd look good if the red and yellow icons were shown within a circle, with the top of the ! protruding through it - as our logo does.

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -100,6 +100,16 @@ public function showIndex()
 
         $canPageBackward = Incident::notScheduled()->where('created_at', '<', $startDate->format('Y-m-d'))->count() != 0;
 
+        // Determine favicon to use
+        $favicon = 'favicon';
+        if (Component::where('status', '=', 4)->count() > 0 ||
```
Comment (jbrooksuk) [inline app/Http/Controllers/HomeController.php]: By default `where` will use the `=` operator, meaning you can just do `Component::where('status', 4)` :)

Comment (jbrooksuk) [inline app/Http/Controllers/HomeController.php]: But to reduce your code down, you could actually do `Component::whereIn('status', [2, 4])->count()`

**Comment by TheoBearman** [comment]
Or that ^

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -100,6 +100,16 @@ public function showIndex()
...
+            Incident::where('status', '=', 2)->count() > 0) {
+            $favicon = 'favicon-high-alert';
+        } elseif (Component::where('status', '>=', 2)->where('status', '<=', 3)->count() > 0 ||
+            Incident::where('status', '=', 1)->orWhere('status', '=', 3)->count() > 0) {
```
Comment (jbrooksuk) [inline app/Http/Controllers/HomeController.php]: Same here, `whereIn` :)

File: app/Http/Controllers/HomeController.php (hunk 1 above)
Comment (GrahamCampbell) [inline app/Http/Controllers/HomeController.php]: Yeh, we don't want multiple queries.

**Comment by DivineOmega** [comment]
Good idea with the icons. I'm no designer, but I've tweaked them to be exclamation marks protruding from a circle:

| Name | Image |
---------|-------
| Medium Alert | favicon-medium-alert |
| High Alert | favicon-high-alert |

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -113,6 +121,7 @@ public function showIndex()
             'previousDate'         => $startDate->copy()->subDays($daysToShow)->toDateString(),
             'nextDate'             => $startDate->copy()->addDays($daysToShow)->toDateString(),
             'pageTitle'            => Setting::get('app_name').' Status',
+            'favicon'              => $favicon,
```
Comment (jbrooksuk) [inline app/Http/Controllers/HomeController.php]: Maybe put this into `IndexComposer`?

Comment (DivineOmega) [inline app/Http/Controllers/HomeController.php]: Good idea. Do you think I should move just the default value or the full logic for determining which favicon to show? Full logic makes sense to me.

Comment (jbrooksuk) [inline app/Http/Controllers/HomeController.php]: Full logic please buddy :)

**Comment by jbrooksuk** [comment]
@DivineOmega ok, I think I'm happy with the conditions on which the favicons are chosen. I noticed we only have the PNG icons. Is it possible to get an SVG version of them too? Once that's done, squash the commits into one and I'll merge :smile:

**Comment by DivineOmega** [comment]
@jbrooksuk I've re-based.

Regarding the SVGs: I didn't originally work from vectors, so I'd have to manually vectorise the PNGs. I may be able to get someone I know to do this as a separate pull request, if necessary?

**Comment by jbrooksuk** [comment]
Always work form vectors :wink: 

I'll merge this later when I'm able to immediately push updated assets :)

**Comment by DivineOmega** [comment]
I should be able to get SVGs tomorrow and the icons redone from those vectors.

**Comment by jbrooksuk** [comment]
Thanks! I wasn't able to do it last night.

**Comment by joecohens** [comment]
I think the icon can be the logo icon in different colours?

**Comment by GrahamCampbell** [comment]
> I think the icon can be the logo icon in different colours?

I'd prefer that.

**Comment by TheoBearman** [comment]
@joecohens @GrahamCampbell That's what I first suggested! ^

**Comment by joecohens** [comment]
@TheoBearman I just saw your comment :) thanks!

**Comment by GrahamCampbell** [comment]
lol

**Comment by jbrooksuk** [comment]
Ok :)

**Comment by jbrooksuk** [comment]
Just pushed the new icons.

**Comment by TheoBearman** [comment]
Nice to see we went with my suggestion in the end. Good job @jbrooksuk.

**Comment by jbrooksuk** [comment]
:+1: good suggestion!

**Comment by DivineOmega** [comment]
Thanks for merging!

I know we're using the Cachet coloured icons, but I do now have the exclamation mark icons done as SVGs (below). Hopefully they'll still be useful? :)

http://imgh.us/exclamation_mark_yellow.svg
http://imgh.us/exclamation_mark_red.svg

**Pull request merged**